File: FinancialMonitoring.Entities/Finance/FinancialItem.cs
using FinancialMonitoring.Entities.Security;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace FinancialMonitoring.Entities.Finance
{
  /// <summary>
  /// A fiancial item
  /// </summary>
  public sealed class FinancialItem : IFinancialItem
  {
    /// <summary>
    /// Id of the item
    /// </summary>
    [JsonProperty("_financialItemId")]
    public Guid FinancialItemId { get; set; }

    /// <summary>
    /// Id of the Category the item belongs to
    /// </summary>
    [JsonProperty("_categoryId")]
    public Guid CategoryId { get; set; }

    /// <summary>
    /// The title of the item
    /// </summary>
    [JsonProperty("_title")]
    public string Title { get; set; }

    /// <summary>
    /// The value of the item
    /// </summary>
    [JsonProperty("_value")]
    public decimal Value { get; set; }

    /// <summary>
    /// Date when the item was due
    /// </summary>
    [JsonProperty("_dueDate")]
    public DateTime DueDate { get; set; }

    ///<inheritdoc/>
    public object Id => FinancialItemId;

    ///<inheritdoc/>
    [JsonProperty("_userId")]
    public Guid UserId { get; set; }

    #region references

    /// <summary>
    /// The user
    /// </summary>
    public User User { get; set; }

    /// <summary>
    /// Category the item belongs to
    /// </summary>
    public FinancialCategory Category { get; set; }

    /// <summary>
    /// Attachements of the item
    /// </summary>
    public List<AttachmentItem> Attachments { get; } = new List<AttachmentItem>();

    #endregion
  }
}

File: WebApplication1/ClientApp/src/app/models/finance/FinancialItem.ts
import { FinancialCategory } from "./FinancialCatgegory";
import { AttachmentItem } from "./AttachmentItem";

/**
 * A fiancial item
 */
export class FinancialItem {

  private _financialItemId: string;
  /**
   * @returns {string} The id of the item
   */
  get financialItemId(): string { return this._financialItemId; }
  /**
   * @param {string} value The id of the item
   */
  set financialItemId(value: string) { this._financialItemId = value; }

  private _categoryId: string;
  /**
   * @returns {string} The id of the category
   */
  get categoryId(): string { return this._categoryId; }
  /**
   * @param {string} value The id of the category
   */
  set categoryId(value: string) { this._categoryId = value; }

  private _title: string;
  /**
   * @returns {string} The title of the item
   */
  get title(): string { return this._title; }
  /**
   * @param {string} value The title of the item
   */
  set title(value: string) { this._title = value; }

  private _value: number;
  /**
   * @returns {number} The value of the item
   */
  get value(): number { return this._value; }
  /**
   * @param {number} value The value of the item
   */
  set value(value: number) { this._value = value; }

  private _dueDate: Date;
  /**
   * @returns {Date} Date when the item was due
   */
  get dueDate(): Date { return this._dueDate; }
  /**
   * @param {Date} value Date when the item was due
   */
  set dueDate(value: Date){ this._dueDate = value; }

  private _userId;
  /**
   * @returns {string} The id of the user
   */
  get userId(): string { return this._userId; }
  /**
   * @param {string} value The id of the user
   */
  set userId(value: string) { this._userId = value; }

  private _category: FinancialCategory;
  /**
   * @returns {FinancialCategory} The category
   */
  get category(): FinancialCategory { return this._category; }
  /**
   * @param {FinancialCategory} value The  category
   */
  set category(value: FinancialCategory) { this._category = value; }

  private _attachments: AttachmentItem[];
  /**
   * @returns {AttachmentItem[]} The attachments to the item
   */
  get attachments(): AttachmentItem[] { return this._attachments; }
  /**
   * @param {AttachmentItem[]} value The attachments to the item
   */
  set attachments(value: AttachmentItem[]) { this._attachments = value; }
}
